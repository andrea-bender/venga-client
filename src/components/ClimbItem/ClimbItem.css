@media only screen and (min-width: 300px) {
  button.delete{
    height: 2.5em;
    background-color: #80808059;
  }
  ul.climb_list{
    margin-bottom: 0;
  }
}