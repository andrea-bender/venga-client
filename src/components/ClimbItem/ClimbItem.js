import React from 'react';
import ClimbsContext from '../../contexts/ClimbsContext';
import config from '../../config';
import './ClimbItem.css';

function deleteClimbRequest(id, cb) {
  fetch(config.API_ENDPOINT + `/climbs/${id}`, {
    method: 'DELETE',
    headers: {
      'content-type': 'application/json'
    }
  })
    .then(res => {
      if(!res.ok) {
        return res.json().then(error => Promise.reject(error))
      }
    })
    .then(() => {
      cb(id)
    })
    .catch(error => {
      console.error(error)
    })
};

export default function ClimbItem(props) {
  return (
    <ClimbsContext.Consumer>
      { (context) => (
        <div className ='climbItem'>
          <div className = 'climbItem_info'>
            <h3>{props.name}</h3>
              
            <ul>
              <li>Location: {props.location}</li>
              <li>Grade: {props.grade} </li>
              <li>Type:{props.type}</li>
              <li>Description: {props.description}</li>
              <li>Rating: {props.rating} Stars</li>
            </ul>
          </div>

          <div className='climbsButton'>
            <button className='delete'
              onClick={() => {
              deleteClimbRequest(
                props.id,
                context.deleteClimb
              )
              }}> 
                Delete Climb
            </button>
          </div>
        </div>
      )}
    </ClimbsContext.Consumer>
  )

};