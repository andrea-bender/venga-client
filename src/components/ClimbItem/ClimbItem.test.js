import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ClimbItem from './ClimbItem'

describe(`ClimbItem component`, () => {
  const props = {
    id: 'a11114yjfj',
    name: 'test-class-name',
    location: 'test-location',
    description: 'test-description',
    grade: 'test-grade',
    type: 'Traditional',
    rating:'4'
  }

  it('renders climbs by default', () => {
    const wrapper = shallow(<ClimbItem />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders climbs given props', () => {
    const wrapper = shallow(<ClimbItem {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})