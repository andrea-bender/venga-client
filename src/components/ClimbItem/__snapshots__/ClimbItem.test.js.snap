// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClimbItem component renders climbs by default 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`ClimbItem component renders climbs given props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
